non-determinism test is failing

I get consistent failures locally when running `make test-sim-nondeterminism` on release/v0.50.x-celestia but not on release/v0.50.x

some of the failures seem unrelated, but also seems like this test should be working. Its quite flakey in CI as well. 

```
Starting the simulation from time Mon Mar 28 17:17:06 UTC 2236 (unixtime 8401684626)
Simulating... block 2/100, operation 0/370. Logs to writing to /home/evan/.simapp/simulations/1751481242497.log
    simulate.go:345: error on block  2/100, operation (36/370) from x/distribution:
�E?{>r': invalid request [/home/evan/go/pkg/mod/cosmossdk.io/errors@v1.0.1/errors.go:155]ing account �Kt�
        Comment: unable to deliver tx
--- FAIL: TestAppStateDeterminism (0.56s)
FAIL
exit status 1
FAIL	cosmossdk.io/simapp	0.641s
make: *** [Makefile:261: test-sim-nondeterminism] Error 1
```

--- a/x/distribution/simulation/operations.go
+++ b/x/distribution/simulation/operations.go
@@ -99,6 +99,11 @@ func SimulateMsgSetWithdrawAddress(txConfig client.TxConfig, ak types.AccountKee
 		simToAccount, _ := simtypes.RandomAcc(r, accs)
 
 		account := ak.GetAccount(ctx, simAccount.Address)
+		_, ok := account.(types.VestingAccount)
+		if ok {
+			return simtypes.NoOpMsg(types.ModuleName, sdk.MsgTypeURL(&types.MsgSetWithdrawAddress{}), "vesting account can't have withdraw addresses since CIP-31"), nil, nil
+		}
+
 		spendable := bk.SpendableCoins(ctx, account.GetAddress())
 
 		msg := types.NewMsgSetWithdrawAddress(simAccount.Address, simToAccount.Address)

